NullPointerException in Geopoint.<init>

From Google Play Console reported for beta 2021.03.16-RC1 but also present in 2021.01.23

416 occurences, 332 users, Android 9,10,11

```
java.lang.RuntimeException: 
  at android.app.ActivityThread.deliverResults (ActivityThread.java:4975)
  at android.app.ActivityThread.handleSendResult (ActivityThread.java:5016)
  at android.app.servertransaction.ActivityResultItem.execute (ActivityResultItem.java:51)
  at android.app.servertransaction.TransactionExecutor.executeCallbacks (TransactionExecutor.java:135)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:95)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2071)
  at android.os.Handler.dispatchMessage (Handler.java:107)
  at android.os.Looper.loop (Looper.java:224)
  at android.app.ActivityThread.main (ActivityThread.java:7561)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:539)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:995)
Caused by: java.lang.NullPointerException: 
  at cgeo.geocaching.location.Geopoint.<init> (Geopoint.java:95)
  at cgeo.geocaching.maps.DistanceDrawer.setCoordinates (DistanceDrawer.java:26)
  at cgeo.geocaching.maps.google.v2.GoogleMapView.setCoordinates (GoogleMapView.java:347)
  at cgeo.geocaching.maps.AbstractMap.setTarget (AbstractMap.java:117)
  at cgeo.geocaching.maps.google.v2.GoogleMapActivity.onActivityResult (GoogleMapActivity.java:211)
  at android.app.Activity.dispatchActivityResult (Activity.java:8249)
  at android.app.ActivityThread.deliverResults (ActivityThread.java:4968)
```

Related to #10071 ?

--- a/main/src/cgeo/geocaching/maps/DistanceDrawer.java
+++ b/main/src/cgeo/geocaching/maps/DistanceDrawer.java
@@ -23,9 +23,12 @@ public class DistanceDrawer {
     }
 
     public void setCoordinates(final Location location) {
-        final Geopoint currentCoords = new Geopoint(location);
-        distance = null != destinationCoords ? currentCoords.distanceTo(destinationCoords) : 0.0f;
-    }
+        if (destinationCoords == null || location == null) {
+            distance = 0.0f;
+        } else {
+            distance = new Geopoint(location).distanceTo(destinationCoords);
+        }
+     }
 
     public Geopoint getDestinationCoords() {
         return destinationCoords;

